File: backend/hackathon/src/main/java/com/dev_trails/hackathon/Policy.java
package com.dev_trails.hackathon;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Column;
@Entity
public class Policy {
@Id
@GeneratedValue(strategy=GenerationType.IDENTITY)
public Long id;
@Column(unique=true)
public String policyNumber;
public Long riderId;
public String riderName;
public String plan;
public String zone;
public String premium;
public Integer riskScore;
public String status;
public String startDate;
public String email;
public String phone;
public String address;
public String birthdate;
public String customerSince;
public String accountTier;
public String delinquencyStatus;
}

File: backend/models/Policy.js
const mongoose = require('mongoose');

const PolicySchema = new mongoose.Schema({
  policy_number: { type: String, unique: true, sparse: true },
  rider_id: { type: mongoose.Schema.Types.Mixed },
  rider_name: { type: String },
  plan: { type: String },
  zone: { type: String },
  premium: { type: String },
  risk_score: { type: Number },
  status: { type: String, default: 'ACTIVE' },
  start_date: { type: String },
  email: { type: String },
  phone: { type: String },
  address: { type: String },
  birthdate: { type: String },
  customer_since: { type: String },
  account_tier: { type: String },
  delinquency_status: { type: String, default: 'CURRENT' },
}, { timestamps: false });

PolicySchema.set('toJSON', { virtuals: true });
PolicySchema.virtual('id').get(function () { return this._id.toHexString(); });

module.exports = mongoose.model('Policy', PolicySchema);
